include/highfive/bits/H5File_misc.hpp

```diff
@@ -65,25 +64,32 @@ inline File::File(const std::string& filename, unsigned openFlags,
             createMode = H5F_ACC_EXCL;
         } else {
             HDF5ErrMapper::ToException<FileException>(
-                std::string("Unable to open file " + _filename));
+                std::string("Unable to open file " + filename));
         }
     }
 
-    if ((_hid = H5Fcreate(_filename.c_str(), createMode, H5P_DEFAULT,
+    if ((_hid = H5Fcreate(filename.c_str(), createMode, H5P_DEFAULT,
                           fileAccessProps.getId())) < 0) {
         HDF5ErrMapper::ToException<FileException>(
-            std::string("Unable to create file " + _filename));
+            std::string("Unable to create file " + filename));
     }
 }
 
-inline const std::string& File::getName() const noexcept {
-    return _filename;
+inline const std::string File::getName() const noexcept {
+    char name[256];
+    ssize_t st = H5Fget_name(_hid, name, 256);
```

Comment:
I think it is ineffective to store additional data that can be retrieved from native hdf5 library. If we store private std::string variable `_filename` then `sizeof(File)` should be about 50 bytes and `sizeof(Group)` takes only 8 bytes. I may be mistaken but I think is some cases, when doing tree like systems based on HighFive, may garbage RAM.
If by `pitfall` you mean the max possibe length of filename then in `H5Utils.hpp` we have `get_name` method wich is widely used across HighFive and there we also limit the max path length by 255 elements.
I agree that 256 symbols maybe not sufficient, but I think HighFive should restrict some max length by some value and this value should be used across whole library. Probably 512 or 1024 elements?

Reply:
I thought that `name` is a temporary variable that will be freed at the end of its scope (this function). So I'd be surprise if its size is a huge problem. But what @ferdonline means is that you could maybe have a fall-back in case 256 is not enough?

Reply:
thank you for explanation. Probably (though I have some doubts) it is possible to get info about link and the size of its "name" and then decide how long should be char array. But anyway I think this may slow down the IO perfomance. As for me  better solution would be setting length of char array through macros. So the default value say 512 but user may change it as it needs (have not tested yet).

Reply:
I remembered that I'd seen it being done robustly here:
https://github.com/BlueBrain/HighFive/blob/90e6316c171861fb925e7f995533114a4d206614/include/highfive/bits/H5Utils.hpp#L280

And actually. Can't you simply use `details::get_name`?